internal/dag/gatewayapi_processor_test.go

```diff
@@ -202,3 +205,377 @@ func TestComputeHosts(t *testing.T) {
+						ForwardTo: httpRouteForwardTo("purplesvc", 80, 1),
+					}}, 1982),
+				httpRoute("basic2",
+					[]gatewayapi_v1alpha1.Hostname{"test.projectcontour.io"}, []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches:   httpRouteMatch(gatewayapi_v1alpha1.PathMatchPrefix, "/blog"),
+						ForwardTo: httpRouteForwardTo("greensvc", 80, 1),
+					}}, 2004),
+			},
+			want: []*gatewayapi_v1alpha1.HTTPRoute{
+				httpRoute("basic3",
+					[]gatewayapi_v1alpha1.Hostname{"test.projectcontour.io"}, []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches:   httpRouteMatch(gatewayapi_v1alpha1.PathMatchPrefix, "/blog"),
+						ForwardTo: httpRouteForwardTo("purplesvc", 80, 1),
+					}}, 1982),
+			},
+		},
+		"same prefixMatch, different match types": {
+			routes: []*gatewayapi_v1alpha1.HTTPRoute{
+				httpRoute("basic",
+					[]gatewayapi_v1alpha1.Hostname{"test.projectcontour.io"}, []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches:   httpRouteMatch(gatewayapi_v1alpha1.PathMatchExact, "/blog"),
+						ForwardTo: httpRouteForwardTo("bluesvc", 80, 1),
+					}}, 2021),
+				httpRoute("basic2",
+					[]gatewayapi_v1alpha1.Hostname{"test.projectcontour.io"}, []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches:   httpRouteMatch(gatewayapi_v1alpha1.PathMatchPrefix, "/blog"),
+						ForwardTo: httpRouteForwardTo("greensvc", 80, 1),
+					}}, 1982),
+			},
+			want: []*gatewayapi_v1alpha1.HTTPRoute{
+				httpRoute("basic",
+					[]gatewayapi_v1alpha1.Hostname{"test.projectcontour.io"}, []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches:   httpRouteMatch(gatewayapi_v1alpha1.PathMatchExact, "/blog"),
+						ForwardTo: httpRouteForwardTo("bluesvc", 80, 1),
+					}}, 2021),
+				httpRoute("basic2",
+					[]gatewayapi_v1alpha1.Hostname{"test.projectcontour.io"}, []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches:   httpRouteMatch(gatewayapi_v1alpha1.PathMatchPrefix, "/blog"),
+						ForwardTo: httpRouteForwardTo("greensvc", 80, 1),
+					}}, 1982),
+			},
+		},
+		"same prefixMatch, same match types, same timestamps": {
+			routes: []*gatewayapi_v1alpha1.HTTPRoute{
+				httpRoute("basic",
+					[]gatewayapi_v1alpha1.Hostname{"test.projectcontour.io"}, []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches:   httpRouteMatch(gatewayapi_v1alpha1.PathMatchPrefix, "/blog"),
+						ForwardTo: httpRouteForwardTo("bluesvc", 80, 1),
+					}}, 2021),
+				httpRoute("different",
+					[]gatewayapi_v1alpha1.Hostname{"test.projectcontour.io"}, []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches:   httpRouteMatch(gatewayapi_v1alpha1.PathMatchPrefix, "/blog"),
+						ForwardTo: httpRouteForwardTo("greensvc", 80, 1),
+					}}, 2021),
+			},
+			want: []*gatewayapi_v1alpha1.HTTPRoute{
+				httpRoute("basic",
+					[]gatewayapi_v1alpha1.Hostname{"test.projectcontour.io"}, []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches:   httpRouteMatch(gatewayapi_v1alpha1.PathMatchPrefix, "/blog"),
+						ForwardTo: httpRouteForwardTo("bluesvc", 80, 1),
+					}}, 2021),
+			},
+		},
```

Comment:
Sorry can you clarify a bit? Just more positive test cases?

Reply:
yeah, i think i was in the context of asking about what happens when you have multiple routes that are exactly the same  (same path/header match and same forward to), is one still "rejected" or are they "mergeable" and was thinking there should be a test for that case, since we have a test for a similar but not the same thing (same path/header match but different forward to)

also maybe separately what happens when a single route has rules that conflict (same path/header match, maybe same or different forward tos)

Reply:
Do we not have tests for when a single route has conflicting rules already?

In terms of extra test cases, I'd like to see at least one testing that the earliest creation time wins correctly. Seems like the `httpRoute` helper can't do that atm, so everything has matching creation timestamp?